src/kirby/components/list/list.component.ts

```diff
@@ -144,6 +147,10 @@ export class ListComponent implements OnChanges {
   onItemSelect(args: any) {
     this.selectedItem = this.listHelper.getSelectedItem(this.items, args);
     this.itemSelect.emit(this.selectedItem);
+
+    if (isIOS) {
+      this.listViewRef.listView.refresh(); // IOS needs this to repaint layout after the selected item has changed
```

Comment:
You can move this call to `refresh()` into the `listHelper.getSelectedItem` function (or create an other function in it).

On native the `args` argument is of type `ListViewEventData` and `args.object` is of type `RadListView`.

In the helpers `.tns` version you will be able to write

```typescript
  getSelectedItem(items: any[], args: ListViewEventData) {
    let selectedItem: any;
    if (args.index >= 0) {
      selectedItem = items[args.index];
    }
    if (args.object.ios) {
      args.object.refresh();
    }
    return selectedItem;
  }
```

I have not tested this code 😄

Reply:
That didn't work because it needs to update after the emit :)